### PR TITLE
[PT FE]: fix aten::embedding realization for integer-like indicies an…

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *fix aten::embedding realization for integer-like indicies and add tests*
 tested on moussaKam/barthez-orangesum-abstract from hf

### Tickets
- *101072*

### CODE REVIEW

File: src/frontends/pytorch/src/op/embedding.cpp [hunk 1]
```diff
@@ -13,10 +14,12 @@ namespace pytorch {
 namespace op {
 
 OutputVector translate_embedding(NodeContext& context) {
+    // aten::embedding(Tensor weight, Tensor indices, SymInt padding_idx=-1, bool scale_grad_by_freq=False, bool
+    // sparse=False) parameters 2, 3 and 4 used only for training case and connected with gradients propagation
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/embedding.cpp:18]: I don't think `sparse` is used only for training. `sparse` means that weight is sparse tensor, not dense. Also `padding_idx` is unclear how it is only for training.

Comment (eaidova) [reply src/frontends/pytorch/src/op/embedding.cpp:18]: according to docs:
sparse ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – If True, gradient w.r.t. weight matrix will be a sparse tensor. See Notes for more details regarding sparse gradients.



Note

Keep in mind that only a limited number of optimizers support sparse gradients: currently it’s optim.SGD (CUDA and CPU), optim.SparseAdam (CUDA and CPU) and optim.Adagrad (CPU)

https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html#torch.nn.Embedding

Comment (eaidova) [reply src/frontends/pytorch/src/op/embedding.cpp:18]: padding_idx ([int](https://docs.python.org/3/library/functions.html#int), optional) – If specified, the entries at padding_idx do not contribute to the gradient; therefore, the embedding vector at padding_idx is not updated during training, i.e. it remains as a fixed “pad”. For a newly constructed Embedding, the embedding vector at padding_idx will default to all zeros, but can be updated to another value to be used as the padding vector.

so it is padding for gradients.

Comment (eaidova) [reply src/frontends/pytorch/src/op/embedding.cpp:18]: other parameters like max_norm and norm_type made on weights, made before aten::embeddings, they are not a part of op

Comment (mvafin) [reply src/frontends/pytorch/src/op/embedding.cpp:18]: Okay, thank you for this analysis, that would be great if you add more detailed comment about why we skip those inputs.

Comment (eaidova) [reply src/frontends/pytorch/src/op/embedding.cpp]: updated

**Review by mvafin — APPROVED** [review]
